sdk/python/feast/data_source.py

```diff
@@ -671,6 +671,13 @@ def __init__(
     ):
         self._bigquery_options = BigQueryOptions(table_ref=table_ref, query=query)
 
+        if not self._table_exists():
+            raise NameError(
```

Comment:
Wouldn't it be cleaner to raise an exception in `_table_exists()`?

Reply:
Good idea, let me make it cleaner.